OptimizelySDK/Optimizely.cs

```diff
@@ -568,22 +644,28 @@ public virtual bool IsFeatureEnabled(string featureKey, string userId, UserAttri
         /// <returns>string | Feature variable value or null</returns>
         public string GetFeatureVariableString(string featureKey, string variableKey, string userId, UserAttributes userAttributes = null)
         {
-            return GetFeatureVariableValueForType<string>(featureKey, variableKey, userId, userAttributes, FeatureVariable.VariableType.STRING);
+            var config = ProjectConfigManager?.GetConfig();
+            if (!IsValid && config == null) {
+                Logger.Log(LogLevel.ERROR, "Datafile has invalid format. Failing 'activate'.");
+                return null;
+            }
+
+            return GetFeatureVariableValueForType<string>(featureKey, variableKey, userId, userAttributes, FeatureVariable.VariableType.STRING, config);
         }
 
         /// <summary>
         /// Sends impression event.
         /// </summary>
         /// <param name="experiment">The experiment</param>
-        /// <param name="variationId">The variation entity</param>
+        /// <param name="variation">The variation entity</param>
         /// <param name="userId">The user ID</param>
         /// <param name="userAttributes">The user's attributes</param>
-        private void SendImpressionEvent(Experiment experiment, Variation variation, string userId, 
-        UserAttributes userAttributes)
+        private void SendImpressionEvent(Experiment experiment, Variation variation, string userId,
+                                         UserAttributes userAttributes, ProjectConfig config)
```

Comment:
We don't need to take in the `ProjectConfig` as a param. We can use the `ProjectConfigManager` in here

Reply:
This method is used by `Activate`, if we use `ProjectConfigManager` inside `SendImpressionEvent`, there are chances `ProjectConfig` used to get `variation` from `DecisionService` may not be same when get from `ProjectConfigManagr` inside this method.

Reply:
Hmm, ok that could actually happen.